GVFS/GVFS/CommandLine/UpgradeVerb.cs

```diff
@@ -0,0 +1,325 @@
+
+            if (this.upgrader.IsNoneRing())
+            {
+                this.tracer.RelatedInfo($"{nameof(this.TryRunProductUpgrade)}: {GVFSConstants.UpgradeVerbMessages.NoneRingConsoleAlert}");
+                this.ReportInfoToConsole(GVFSConstants.UpgradeVerbMessages.NoneRingConsoleAlert);
+                this.ReportInfoToConsole(GVFSConstants.UpgradeVerbMessages.SetUpgradeRingCommand);
+                return true;
+            }
+
+            if (!this.TryRunUpgradeChecks(out newestVersion, out isInstallable, out error))
+            {
+                this.Output.WriteLine(errorOutputFormat, error);
+                this.tracer.RelatedError($"{nameof(this.TryRunProductUpgrade)}: Upgrade checks failed. {error}");
+                return false;
+            }
+
+            if (newestVersion == null)
+            {
+                this.ReportInfoToConsole($"Great news, you're all caught up on upgrades in the {this.upgrader.Ring} ring!");
+                return true;
+            }
+
+            this.ReportInfoToConsole("New GVFS version available: {0}", newestVersion.ToString());
+                        
+            if (!this.Confirmed && isInstallable)
+            {
+                this.ReportInfoToConsole("Run `gvfs upgrade --confirm` to install it");
+                return true;
+            }
+
+            if (!isInstallable)
+            {
+                this.tracer.RelatedError($"{nameof(this.TryRunProductUpgrade)}: {error}");
+                this.Output.WriteLine(errorOutputFormat, error);
+                return false;
+            }
+
+            if (!this.TryRunInstaller(out error))
+            {
+                this.tracer.RelatedError($"{nameof(this.TryRunProductUpgrade)}: Could not launch upgrade tool. {error}");
+                this.Output.WriteLine(errorOutputFormat, "Could not launch upgrade tool. " + error);
+                return false;
+            }
+
+            return true;
+        }
+
+        private bool TryRunUpgradeChecks(
+            out Version latestVersion,
+            out bool isUpgradeInstallable,
+            out string consoleError)
+        {
+            bool upgradeCheckSuccess = false;
+            bool upgradeIsInstallable = false;
+            string errorMessage = null;
+            Version version = null;
+
+            this.ShowStatusWhileRunning(
+                () =>
+                {
+                    upgradeCheckSuccess = this.TryCheckUpgradeAvailable(out version, out errorMessage);
+                    if (upgradeCheckSuccess && version != null)
+                    {
+                        upgradeIsInstallable = true;
+
+                        if (!this.TryCheckUpgradeInstallable(out errorMessage))
+                        {
+                            upgradeIsInstallable = false;
+                        }
+                    }
+
+                    return upgradeCheckSuccess;
+                },
+                 "Checking for GVFS upgrades",

```

Comment:
Elsewhere you've already started calling it "VFS for Git". Please make it consistent.

Reply:
There are too many occurrences of GVFS, so renamed "VFS for Git" to "GVFS". Will replace "GVFS" references with "VFS for Git" in a future PR.